core/src/main/java/org/apache/carbondata/core/reader/CarbonFooterReaderV3.java

```diff
@@ -59,17 +55,20 @@ public FileFooter3 readFooterVersion3() throws IOException {
     thriftReader.open();
 
     // If footer offset is 0, means caller does not set it,
-    // so we read it from the end of the file
+    // so we read it from the index file
     if (footerOffset == 0) {
```

Comment:
The function of this method should be single, it should only support reading footer.

Reply:
if it needs to read the index, better to implement in another place where it decides to read index or footer.